### PR TITLE
Tied schema dropdown to request/response schema samples

### PR DESCRIPTION
**Pull request by craiglarrabee**
I needed the json schema samples to change based on the selection of a schema descendant in the dropdown. This change does not affect samples provided directly in the swagger.json.

### CODE REVIEW

**Comment by craiglarrabee** [comment]
There is a failing check on this, but from what I can tell, it doesn't have anything to do with what I submitted. Will this be reviewed?

**Comment by RomanGotsiy** [comment]
Hey @craiglarrabee, thanks for this awesome PR!
I will review it ASAP

**Comment by craiglarrabee** [comment]
Thanks @RomanGotsiy. I hope it's something you can use.

**Comment by michaelsbradleyjr** [comment]
Any update on when this might get reviewed–merged? I need this as well.

**Comment by michaelsbradleyjr** [comment]
Actually, I tried running `npm run-script build:all` for both @craiglarrabee's master branch and also when rebasing his branch against Rebilly/master.

In both cases, I had to add `import { Observable } from 'rxjs/Observable'` to `lib/services/schema-changer.service.ts` in order for the compile step to not fail.

And, in both cases, after successful build, I don't see that any changes are trigger in the JSON Request Samples nor the Response Samples when a descendant is toggled in the dropdown.

`x-extendedDiscriminator` is really nice! BUT, if it's not possible for the displayed samples (in my case I'm interested in Response Samples) to be updated via the dropdown, it amounts to an incomplete feature. Maybe I'm doing something wrong?

**Comment by craiglarrabee** [comment]
@michaelsbradleyjr, is your dropdown (type selector) on the top level object, or on a field represented by an object?

**Comment by michaelsbradleyjr** [comment]
On a top-level field/key in the response schema.

**Comment by craiglarrabee** [comment]
I have not used `extendedDiscriminator`. I'll have to look into what's happening with that.

**Comment by craiglarrabee** [comment]
@michaelsbradleyjr I've added the support for x-extendedDiscriminator. Hope that works for you now. I wasn't aware of that custom extension, and it's very useful, so thanks for bringing it to my attention.

**Comment by michaelsbradleyjr** [comment]
Thanks for looking into it and adding support!  I'll give it a whirl early next week, no time this weekend, else I would try soon.

**Comment by RomanGotsiy** [comment]
Hey @michaelsbradleyjr, sorry for not merging this for so long time 🙈 ...
I am working on React rewrite of ReDoc (2.0) so I'm quite busy. But I think we should merge this change so users of 1.x version can benefit from this.

I will try to review it and merge early next week.
Thanks for contributing!

File: lib/components/JsonSchema/json-schema.ts [hunk 1]
```diff
@@ -55,6 +56,9 @@ export class JsonSchema extends BaseSearchableComponent implements OnInit {
 
   selectDescendantByIdx(idx) {
     this.selectDescendant(this.descendants[idx]);
+    if (this.descendants[idx]) {
+      this._schemaChanger.announceDescendantChange(idx, this.descendants[idx].name, this.isRequestSchema ? true : false, this.responseCode);
```
Comment (simonrichard) [inline lib/components/JsonSchema/json-schema.ts]: `this.isRequestSchema` is already a boolean, so you could get rid of the ternary operator.

**Comment by RomanHotsiy** [comment]
This PR is opened against v1.x branch.
Unfortunately, v1.x is now in the maintenance mode and no new features will be added.

@craiglarrabee thanks for your amazing contribution and terribly sorry for not merging this earlier

If somebody is still interested in this feature you can open a similar PR against the latest version of Redoc. I promise this time it won't take ages before merged.

**Pull request closed without merging**